From 70c71908f6db88826ba74de1d5aa1adc3871f0ba Mon Sep 17 00:00:00 2001
From: Frode Nordahl <fnordahl@ubuntu.com>
Date: Thu, 24 Oct 2024 17:46:14 +0200
Subject: [PATCH 4/6] route-table: Expose route_table_parse.

From: Felix Huettner <felix.huettner@stackit.cloud>

also used on the OVN side to watch for routing updates.

Signed-off-by: Felix Huettner <felix.huettner@stackit.cloud>
---
 lib/route-table.c | 3 +--
 lib/route-table.h | 3 ++-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/route-table.c b/lib/route-table.c
index 918e86472..9b5c9f29d 100644
--- a/lib/route-table.c
+++ b/lib/route-table.c
@@ -65,7 +65,6 @@ static bool route_table_valid = false;
 static void route_table_reset(void);
 static void route_table_handle_msg(const struct route_table_msg *,
                                    void *);
-static int route_table_parse(struct ofpbuf *, void *change);
 static void route_table_change(const struct route_table_msg *, void *);
 static void route_map_clear(void);
 
@@ -210,7 +209,7 @@ route_table_reset(void)
 
 /* Return RTNLGRP_IPV4_ROUTE or RTNLGRP_IPV6_ROUTE on success, 0 on parse
  * error. */
-static int
+int
 route_table_parse(struct ofpbuf *buf, void *change_)
 {
     struct route_table_msg *change = change_;
diff --git a/lib/route-table.h b/lib/route-table.h
index 4f44a8085..968652d66 100644
--- a/lib/route-table.h
+++ b/lib/route-table.h
@@ -24,7 +24,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-#include "openvswitch/types.h"
+#include "openvswitch/ofpbuf.h"
 
 struct route_data {
     /* Copied from struct rtmsg. */
@@ -55,6 +55,7 @@ uint64_t route_table_get_change_seq(void);
 void route_table_init(void);
 void route_table_run(void);
 void route_table_wait(void);
+int route_table_parse(struct ofpbuf *buf, void *change_);
 bool route_table_fallback_lookup(const struct in6_addr *ip6_dst,
                                  char name[],
                                  struct in6_addr *gw6);
-- 
2.45.2

